utils/rate_test.go

```diff
@@ -242,6 +258,19 @@ func (r *runnerImpl) assertCountAt(d time.Duration, count int) {
 	})
 }
 
+// assertCountAtWithNoise like assertCountAt but also considers possible noise in CI
+func (r *runnerImpl) assertCountAtWithNoise(d time.Duration, count int, noise int) {
+	r.wg.Add(1)
+	r.afterFunc(d, func() {
+		actual := int(r.count.Load())
+		boundMin, boundMax := count-noise, count+noise
+		withinRange := boundMin < actual && actual < boundMax
+		assert.True(r.t, withinRange, "expected count to be within [%d, %d], but got %d",
```

Comment:
Is this using `assert` because of the original package? We use `require` which has functions with check for things within range etc. natively.

Reply:
yes, the original tests were already written with assert. That's good to know, I'll go ahead and replace other `assert` with `require` as well

Reply:
replaced them